ament_tools/build_types/ament_cmake.py

```diff
@@ -107,14 +113,38 @@ def on_build(self, context):
                 cmake_args += ["-DAMENT_ENABLE_TESTING=1"]
             if context.symlink_install:
                 cmake_args += ['-DAMENT_CMAKE_SYMLINK_INSTALL=1']
+            if IS_WINDOWS:
+                vsv = get_visual_studio_version()
+                if vsv is None:
+                    print("VisualStudioVersion is not set, please run within "
+                          "a VS2013 or VS2015 Command Prompt.")
+                    raise VerbExecutionError(
+                        "Could not determine Visual Studio Version.")
+                generator = None
+                if vsv == '12.0':
+                    generator = 'Visual Studio 12 2013 Win64'
+                elif vsv == '14.0':
+                    generator = 'Visual Studio 14 2015 Win64'
+                else:
+                    raise VerbExecutionError("Unknown VS version: " + vsv)
+                cmake_args += ['-G', generator]
             if CMAKE_EXECUTABLE is None:
                 raise VerbExecutionError("Could not find 'cmake' executable")
             yield BuildAction(prefix + [CMAKE_EXECUTABLE] + cmake_args)
-        else:
+        elif not IS_WINDOWS:  # Nothing to do if False on Windows
```

Comment:
I find the comment a bit confusing. After reading the code its clear what it means. What about:

    check for reconfigure where available